src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -50,6 +50,12 @@ object ScalafixPlugin extends AutoPlugin {
           "but can be disabled with `scalafixCaching := false`."
       )
 
+    val scalafixArgsOnCompile: TaskKey[Seq[String]] =
+      taskKey[Seq[String]](
+        "Pass extra arguments to scalafix only when scalafix is triggred by compilation.  " +
+          "Only --triggered argument is passed by default."
+      )
```

Comment:
`scalafixArgsOnCompile` TaskKey is intended to be more extensible than just passing `--triggered`.

For example:
 - `scalafixArgsOnCompile += "--check"` allows to lint on compilation and rewrite on explicit invocation in the same `.scalafix.conf`.
 - Preferring `TaskKey` over `SettingKey` because `TaskKey` can rely on other `TaskKey` values, while `scalafixArgsOnCompile` is rarely referred by other `SettingKey`s. (Analogous to `scalacOptions`)

Reply:
See my comment in https://github.com/scalacenter/scalafix/pull/1217#pullrequestreview-506174142, I am questionning the need for such a "open" customization point if it's just for `--check`. Do you have something else in mind?

Reply:
I'd think several alternatives of arguments on compilation in my mind.

1. `--triggered --check` : This should work well in most use cases, but it might be confusing if `.scalafix.conf` gets bigger and a lot of difference in configuration between `triggered` and explicit invocation.

2. `--config .scalafix.oncompile.conf` : In case `.scalafix.conf` is already big, or overlaying settings will make huge difference between on-compile or not, it would be easier to divide-and-conquer scalafix.conf file rather than `overlaying`.

3. `--check` : This makes `scalafixOnCompile` less powerful, but more simple. I'd think this is just a matter of preference, so I wouldn't push this use case so much.

Disclaimer: these alternatives are based on my assumptions. While I haven't seen **HUGE** `.scalafix.conf` which has to be divided and conquered for now, it might be chicken or egg.

Reply:
I have reconsidered this customization point, and removed it for now.
There are few rules which may break source codes in running on-compile, and can be managed only with `--triggered`.